src/SprykerSdk/Zed/AopSdk/Business/OpenApi/Builder/OpenApiCodeBuilder.php

```diff
@@ -0,0 +1,732 @@
+            }
+            if ($mediaType->schema instanceof Reference) {
+                $requestBodyProperties[$this->getClassNameFromReference($mediaType->schema)] = $this->getRequestBodyPropertiesFromReference($mediaType->schema);
+            }
+        }
+
+        return $requestBodyProperties;
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Operation $operation
+     *
+     * @return iterable
+     */
+    protected function getRequestBodyFromOperation(Operation $operation): iterable
+    {
+        return isset($operation->requestBody) && isset($operation->requestBody->content) ? $operation->requestBody->content : [];
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Schema $schema
+     *
+     * @return iterable
+     */
+    protected function getPropertiesFromSchema(Schema $schema): iterable
+    {
+        if (isset($schema->properties)) {
+            return $schema->properties;
+        }
+
+        return [];
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Reference $reference
+     *
+     * @return iterable
+     */
+    protected function getPropertiesFromReference(Reference $reference): iterable
+    {
+        if (isset($reference->properties)) {
+            return $reference->properties;
+        }
+
+        return [];
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Schema $schema
+     *
+     * @return array
+     */
+    protected function getRequestBodyPropertiesFromSchema(Schema $schema): array
+    {
+        foreach ($this->getPropertiesFromSchema($schema) as $schemaObject) {
+            if (!isset(($schemaObject->properties))) {
+                continue;
+            }
+
+            if (empty($schemaObject->properties)) {
+                continue;
+            }
+
+            if ($schemaObject instanceof Schema) {
+                return $this->getRequestBodyPropertiesFromSchema($schemaObject);
+            }
+            if ($schemaObject instanceof Reference) {
+                return $this->getRequestBodyPropertiesFromReference($schemaObject);
+            }
+        }
+
+        return $this->formatRequestBodyProperties($this->getPropertiesFromSchema($schema));
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Reference $schema
+     *
+     * @return array
+     */
+    protected function getRequestBodyPropertiesFromReference(Reference $schema): array
+    {
+        foreach ($this->getPropertiesFromReference($schema) as $schemaObject) {
+            if (!isset(($schemaObject->properties))) {
+                continue;
+            }
+
+            if (empty($schemaObject->properties)) {
+                continue;
+            }
+
+            if ($schemaObject instanceof Schema) {
+                return $this->getRequestBodyPropertiesFromSchema($schemaObject);
+            }
+
+            if ($schemaObject instanceof Reference) {
+                return $this->getRequestBodyPropertiesFromReference($schemaObject);
+            }
```

Comment:
```suggestion
               return $this->getRequestBodyPropertiesFromSchemaOrReference($schemaOrReference);
            
```

Isn't there an interface or something that is implemented by both types?